### ISSUE TITLE
Influence of barcodes/adapters

### ISSUE THREAD

**Issue by DRL** [KamilSJaron/smudgeplot#3]
Hi guys, 

does adapter contamination or 10X barcode contamination/leftover matter?

cheers, 

dom

**Comment by KamilSJaron** [comment]
Good question. I suspect it should be alright. I expect adapter contamination to have incredible coverage and that we are analysing kmers with a upper coverage threshold (the parameter U in dumping kmeres for the analysis)

```
jellyfish dump -c -L 40 -U 600 kmer_counts.jf | hetkmers.py -o kmer_pairs
```

TODO: should be checked out.

**Comment by KamilSJaron** [comment]
I have run now couple of genomes without trimming of adapters and does not seem to be an issue at all! (I will leave it open before I will formally check it with a dataset that has loads of adapters)

**Comment by KamilSJaron** [comment]
Dom, I should never underestimate your suggestions in the future. Thought that I have good enough evidence that trimming does not matter, but... you were right! Bit more messy datasets do have that much more mistakes by the ends of reads that trimming can help tons to reduce signal-to-noise ratio. I am still not sure if it's trimming of low-quality bases or adapters that change it, but that is not the point here. The point is that trimming matters for low coverage datasets.

TLDR:
1. with sufficient coverage/quality of sequencing it does not matter (i.e. for >100x you can omit trimming just to gain speed)
2. trimming seems to help. Everybody should trim the reads before assembly anyway, hence running this analysis on the trimmed dataset is always a good idea.

This is genomescope, but I bet the reflection to smudgeplot will be about the same:
w/o trimming
![kmc_plot](https://user-images.githubusercontent.com/8181573/57388396-6097e800-71b8-11e9-8193-90dac70e4067.png)
after trimming
![kmc_trimmed_plot](https://user-images.githubusercontent.com/8181573/57388397-6097e800-71b8-11e9-9dbe-69f52db2953a.png)
